OOP Basic task was done

noleynik29 opened the pull request:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine. In those classes I have overrided doWork(), so it will print a message that a certain machine started its work. I have overrided stopWork() as well. It should print messages that certain machines stopped working. In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer is working");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Bulldozer has started working, e.g., "Bulldozer started working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Bulldozer is stopped");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Bulldozer has stopped working, e.g., "Bulldozer stopped working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine{
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Bulldozer is working");
+    }
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void doWork() {
+        System.out.println("Excavator is working");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Excavator has started working, e.g., "Excavator started working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Excavator is stopped");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Excavator has stopped working, e.g., "Excavator stopped working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Excavator extends Machine{
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Excavator is working");
+    }
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public class Machine {
```
This violates the requirement: "Create an abstract class Machine that declares two methods: public abstract void doWork(); public abstract void stopWork();" The class should be declared as abstract.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public class Machine {
+    public void doWork() {
```
This violates the requirement: "Create an abstract class Machine that declares two methods: public abstract void doWork(); public abstract void stopWork();" The method should be declared as abstract and should not have a body.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+
+    }
+
+    public void stopWork() {
```
This violates the requirement: "Create an abstract class Machine that declares two methods: public abstract void doWork(); public abstract void stopWork();" The method should be declared as abstract and should not have a body.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+
+public class Machine {
+    public void doWork() {
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+    public void doWork() {
+
+    }
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,11 @@
+    }
+
+    public void stopWork() {
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,12 @@
 package core.basesyntax;
 
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void doWork() {
+        System.out.println("Truck is working");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Truck has started working, e.g., "Truck started working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Truck is stopped");
```
This violates checklist item: "Write informative messages in methods." The message should clearly indicate that the Truck has stopped working, e.g., "Truck stopped working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Truck extends Machine{
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Truck is working");
+    }
+
```
This violates checklist item: "Please don't add redundant empty lines to your code." There is a redundant empty line after the method signature.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

**Decision:** REQUEST_CHANGES

Your project has the right structure and demonstrates inheritance and polymorphism well, but there are a few critical blockers that must be fixed to meet the core requirements and checklist.

**Critical issues to fix (blockers):**
1. Machine must be abstract and its methods abstract (requirement violation)
   - File: src/main/java/core/basesyntax/Machine.java
   - Issue: Class is not declared `abstract` (line 1), and `doWork()`/`stopWork()` are not `abstract` and have bodies (lines where these methods are declared).
   - Why: The requirement states: “Create an abstract class Machine that declares two methods: `public abstract void doWork();` `public abstract void stopWork();`”. Abstract methods enforce implementation in subclasses.
   - How to fix: Declare the class as `public abstract class Machine` and change method signatures to `public abstract void doWork();` and `public abstract void stopWork();` (remove method bodies).

2. Messages must be informative (checklist: “Write informative messages in methods.”)
   - Files: 
     - src/main/java/core/basesyntax/Bulldozer.java: `doWork()` and `stopWork()`
     - src/main/java/core/basesyntax/Excavator.java: `doWork()` and `stopWork()`
     - src/main/java/core/basesyntax/Truck.java: `doWork()` and `stopWork()`
   - Issue: Current print statements don’t clearly indicate start/stop actions for each machine type.
   - Why: Output must clearly reflect the machine type and action.
   - How to fix: Use messages like: “Bulldozer started working” / “Bulldozer stopped working”, “Excavator started working” / “Excavator stopped working”, “Truck started working” / “Truck stopped working”.

**Minor/style improvements (non-blockers but please fix):**
- Remove redundant empty lines (checklist: “Please don't add redundant empty lines to your code.”)
  - Locations:
    - After class declarations: Bulldozer.java, Excavator.java, Truck.java, Machine.java, MainApp.java
    - After method signatures: Bulldozer.java, Excavator.java, Truck.java, Machine.java
  - How: Keep a single blank line where appropriate, but avoid extra empty lines immediately after class declarations or method signatures.

You’ve done well setting up the class hierarchy and demonstrating polymorphism in MainApp. After making the above changes, your solution should fully meet the requirements. Keep it up!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

Elena-Bruyako commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
```
remove redundant empty line
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
```
remove redundant empty line
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,12 @@
 
 public class MainApp {
+    public static void main(String[] args) {
+        Machine[] machines = { new Excavator(), new Bulldozer(), new Truck()};
```
```suggestion
        Machine[] machines = {new Excavator(), new Bulldozer(), new Truck()};
```

Elena-Bruyako commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Truck extends Machine {
+
```
remove redundant empty line
```suggestion
```

Elena-Bruyako reviewed (changes requested):
Good job! Just some style mistakes

noleynik29 committed:
Created class Machine containing methods public abstract void doWork() and public abstract void stopWork().
Created classes Truck, Bulldozer, and Excavator that have inherited from Machine.
In those classes I have overrided doWork(), so it will print a message that a certain machine started its work.
I have overrided stopWork() as well. It should print messages that certain machines stopped working.
In the MainApp class created a Machine array with Truck, Bulldozer, and Excavator and called methods doWork() and stopWork() in a loop.

Elena-Bruyako reviewed (approved):
Good job
